### PR TITLE
Chicoma DiffLodeRunner DDP Harness

### PR DESCRIPTION
**Pull request by skylarjean**
Despite the name of the branch, this is actually a harness for Chicoma. I have tested that the functions to train an epoch run on Chicoma. I have not tested the whole harness with slurm submission etc

### CODE REVIEW

File: src/yoke/utils/training/epoch/loderunner.py [hunk 1]
```diff
@@ -8,13 +8,8 @@
 from yoke.utils.training.datastep.loderunner import (
     train_loderunner_datastep,
     eval_loderunner_datastep,
-    eval_loderunner_datastep_cylex,
```
Comment (hickmank) [inline src/yoke/utils/training/epoch/loderunner.py:11]: Need to leave these. They are "possible" imports based on what training function is requested.

File: src/yoke/utils/training/datastep/loderunner.py [hunk 2]
```diff
@@ -235,17 +235,16 @@ def train_DDP_loderunner_datastep_cylex(
...
-        Args:
-        data (tuple): tuple of model input, corresponding ground truth, and lead time
-        model (loaded pytorch model): model to train
-        optimizer (torch.optim): optimizer for training set
-        loss_fn (torch.nn Loss Function): loss function for training set
-        device (torch.device): device index to select
-        rank (int): Rank of device
-        world_size (int): Number of total DDP processes
+    Args:
+    data (tuple): tuple of model input, corresponding ground truth, and lead time
```
Comment (hickmank) [inline src/yoke/utils/training/datastep/loderunner.py]: Under `Args:` indent 4-spaces.

File: src/yoke/utils/training/datastep/diff_loderunner.py [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+    tau = tau.to(torch.float32).to(device, non_blocking=True)
+
+    # Make channel indices for all 8 variables (0-7) for both input and output
+    vars = torch.tensor([0, 1, 2, 3, 4, 5, 6, 7]).to(device, non_blocking=True)
+
+    # Forward pass: predict noise
+    noise_pred = model(
+        x=x,
+        y_tau=y_tau,
+        in_vars=vars,
```
Comment (hickmank) [inline src/yoke/utils/training/datastep/diff_loderunner.py:131]: Use the `in/out_vars` you pass in.

File: src/yoke/utils/training/datastep/diff_loderunner.py [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+
+    # Move data to device
+    x = x.to(device, non_blocking=True)
+    y_tau = y_tau.to(device, non_blocking=True)
+    noise = noise.to(device, non_blocking=True)
+    lead_times = lead_times.to(torch.float32).to(device, non_blocking=True)
+    tau = tau.to(torch.float32).to(device, non_blocking=True)
+
+    # Make channel indices for all 8 variables (0-7) for both input and output
+    vars = torch.tensor([0, 1, 2, 3, 4, 5, 6, 7]).to(device, non_blocking=True)
```
Comment (hickmank) [inline src/yoke/utils/training/datastep/diff_loderunner.py:125]: Delete this line.

File: applications/harnesses/ch_DDP_diffLDR/training_START.slurm [hunk 5]
```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+# This is a setup for GPU training on Venado. Find out how much
```
Comment (hickmank) [inline applications/harnesses/ch_DDP_diffLDR/training_START.slurm]: Could delete this header...

**Review by hickmank — CHANGES_REQUESTED** [review]
Some minor comments. 

Include diffusion model inputs in the harness input files. Include a CSV for setting up the diffusion training runs.

**Comment by skylarjean** [comment]
I removed "num_diffusion_steps" argument because it's not actually used in training/validation (which computes loss from the predicted noise). It is only used in DDIM

**Comment by skylarjean** [comment]
I'm sorry that coverage went down, but I think writing tests for the harnesses is way more work than it's worth since we can't run the tests on the machine we're designing for.

**Comment by skylarjean** [comment]
As far as I know, I have resolved all the comments/concerns we discussed, and this should be good to merge?

**Review by hickmank — APPROVED** [review]
